src/Data/Nat/Properties.agda

```diff
@@ -2101,31 +2101,43 @@ n≤′m+n (suc m) n = ≤′-step (n≤′m+n m n)
 -- Properties of _≤″_ and _<″_
 ------------------------------------------------------------------------
 
-m<ᵇn⇒1+m+[n-1+m]≡n : ∀ m n → T (m <ᵇ n) → suc m + (n ∸ suc m) ≡ n
-m<ᵇn⇒1+m+[n-1+m]≡n m n lt = m+[n∸m]≡n (<ᵇ⇒< m n lt)
+-- equivalence of  _≤″_ to _≤_
 
-m<ᵇ1+m+n : ∀ m {n} → T (m <ᵇ suc (m + n))
-m<ᵇ1+m+n m = <⇒<ᵇ (m≤m+n (suc m) _)
+≤⇒≤″ : _≤_ ⇒ _≤″_
+≤⇒≤″ = less-than-or-equal ∘ m+[n∸m]≡n
 
-<ᵇ⇒<″ : T (m <ᵇ n) → m <″ n
-<ᵇ⇒<″ {m} {n} leq = less-than-or-equal (m+[n∸m]≡n (<ᵇ⇒< m n leq))
+<⇒<″ : _<_ ⇒ _<″_
+<⇒<″ = ≤⇒≤″
 
-<″⇒<ᵇ : ∀ {m n} → m <″ n → T (m <ᵇ n)
-<″⇒<ᵇ {m} (<″-offset k) = <⇒<ᵇ (m≤m+n (suc m) k)
+≤″⇒≤ : _≤″_ ⇒ _≤_
+≤″⇒≤ (≤″-offset k) = m≤m+n _ k
 
 -- equivalence to the old definition of _≤″_
 
-≤″-proof : ∀ {m n} (le : m ≤″ n) → let less-than-or-equal {k} _ = le in m + k ≡ n
+≤″-proof : (le : m ≤″ n) → let less-than-or-equal {k} _ = le in m + k ≡ n
 ≤″-proof (less-than-or-equal prf) = prf
 
--- equivalence to _≤_
+-- yielding corresponding proof and `pattern` aliases for _≤″_/_<_
 
-≤″⇒≤ : _≤″_ ⇒ _≤_
-≤″⇒≤ {zero}  (≤″-offset k) = z≤n {k}
-≤″⇒≤ {suc m} (≤″-offset k) = s≤s (≤″⇒≤ (≤″-offset k))
+≤-proof : m ≤ n → ∃ λ k → m + k ≡ n
```

Comment:
The name for `≤″-proof` was chosen because it extracted the proof component from the record. However, that's not the case here. Can we come up with a better/more standard name for this lemma?  Perhaps `m≤n⇒m+o≡n`?

Reply:
Agree, it's a better name. does it need an `∃` in it? Eg as `m≤n⇒∃[o]m+o≡n`?

Reply:
Yup, good idea.

Reply:
Where would we now use this over `m+[n∸m]≡n : m ≤ n → m + (n ∸ m) ≡ n`, possibly in combination with `-,_`?

Reply:
Fair question. But I think the uses in `Data.Vec.Bounded.Base` still need to match on the equality being `refl`... but perhaps those, too are eliminable. Currently, I'm weeding out *all* the pattern synonyms I introduced for `_≤″_`... they really are all redundant now (and `less-than-or-equal` had only been retained previously for backwards compatibility, but now it can go, I think?)

Reply:
I think that you might be able to refactor `Data.Vec.Bounded.Base.{padRight|padLeft|padBoth}` further, but I think let's in any case stop here for the time being?